lib/OpenQA/Schema/Schema.pm

```diff
@@ -24,12 +25,22 @@ use FindBin qw($Bin);
 # after bumping the version please look at the instructions in the docs/Contributing.asciidoc file
 # on what scripts should be run and how
 our $VERSION = 24;
+our $configname="$Bin/../lib/database.ini";
 
 __PACKAGE__->load_namespaces;
 
-my @paths = ( "$Bin/../lib/database", "$Bin/../../lib/database" );
-unshift(@paths, dirname($ENV{OPENQA_CONFIG}).'/database') if ($ENV{OPENQA_CONFIG});
-__PACKAGE__->config_paths(\@paths);
+
+sub connect_db {
+    my $mode = shift || $ENV{OPENQA_DATABASE} || 'production';
+    CORE::state $schema;
+    unless ($schema) {
+        my %ini;
+        #        print("ENV{OPENQA_DATABASE_CONFIG}=$ENV{OPENQA_DATABASE_CONFIG}\n");
+        tie %ini, 'Config::IniFiles', ( -file => $ENV{OPENQA_DATABASE_CONFIG} || $configname );
+        $schema=__PACKAGE__->connect($ini{$mode});
```

Comment:
>copying, or more likely linking to t/data is needed because of t/data/client.conf. And since all cfg files have to have same directory path now, they must be present there too.

Yeah, but is it needed? I don't think they are used anywhere in tests.

> Second comment: I did not say same dirname, but same config name. While you can specify whole pathname (dir+file name) in OPENQA_CONFIG for openqa.ini, the other files have fixed names (defaults, and same directory as OPENQA_CONFIG). So I think it would be better to have all configs on same standing. 

Aha, I understood that as a same name (ie. everything called openqa.ini) which didn't make sense to me. Fixed name it is then.

> Deciding at runtime seems possible, but not so nice, since one have to check whether OPENQA_CONFIG is file or directory everytime any config file needs to be loaded.

AFAIK they are loaded once on startup, I don't see where the problem is.